bike-erp/backend/src/main/services/authenticationService/AuthenticationService.ts

```diff
@@ -174,12 +173,42 @@ export const authenticateToken = (req, res, next) => {
       return res.status(403).send({ message: "Invalid token" });
     }
 
-    // Verifying user role
-    if (!authorizeRequest(req.originalUrl, user.role)) {
-      return res.status(401).send({ message: "User role not authorized" });
-    }
-
+    // Returning user email
     req.user = user;
+
+    // Calling route function
     next();
   });
 };
+
+// Method to verify user role
+export const verifyRole = (allowedRoles: Role[]) => {
+  return async (req, res, next) => {
+    // Setting the endpoint header to authorization
+    const authHeader = req.headers["authorization"];
+
+    // Setting token header
+    const token = authHeader && authHeader.split(" ")[1];
+
+    // Verifying if the allowedRoles array is undefined
+    if (allowedRoles === undefined) {
+      // Returning error message
+      return res.status(400).send({ message: "No roles provided" });
```

Comment:
I don't think you need to handle this error. because this is not a user error but an implementation error. 
You could still handle it but it shouldn't be a response error but an internal error (implementation)